File: ios/bridge/ExpoBridgeRecommendations.swift
import Foundation

@available(iOS 16.0, *)
enum ExpoBridgeRecommendations {
  static func getRecommendations(
    service: RecommendationsService,
    musicUserToken: String,
    ids: [String]?
  ) async throws -> [String: Any] {
    try await ExpoBridge.asyncBridge {
      let recommendations = try await service.getRecommendations(
        musicUserToken: musicUserToken,
        ids: ids)
      return BridgeResponses.recommendations(recommendations)
    }
  }

  static func getReplay(
    service: RecommendationsService,
    musicUserToken: String,
    year: Int?
  ) async throws -> [String: Any] {
    try await ExpoBridge.asyncBridge {
      let summaries = try await service.getReplay(musicUserToken: musicUserToken, year: year)
      return BridgeResponses.replaySummaries(summaries)
    }
  }
}

File: android/src/main/java/expo/modules/applemusic/bridge/ExpoBridgeRecommendations.kt
package expo.modules.applemusic.bridge

import expo.modules.applemusic.BridgeResponses
import expo.modules.applemusic.RecommendationsRestClient
import expo.modules.kotlin.functions.Coroutine
import expo.modules.kotlin.modules.ModuleDefinitionBuilder

internal fun ModuleDefinitionBuilder.registerRecommendationsBridge(
  recommendations: () -> RecommendationsRestClient,
) {
  AsyncFunction("getRecommendations") Coroutine { musicUserToken: String, ids: List<String>? ->
    BridgeResponses.recommendations(recommendations().getRecommendations(musicUserToken, ids))
  }

  AsyncFunction("getReplay") Coroutine { musicUserToken: String, year: Int? ->
    BridgeResponses.replaySummaries(recommendations().getReplay(musicUserToken, year))
  }
}
